adapters/handlers/rest/db_users/handlers_db_users.go

```diff
@@ -212,6 +235,69 @@ func (h *dynUserHandler) getUser(params users.GetUserInfoParams, principal *mode
 	return users.NewGetUserInfoOK().WithPayload(response)
 }
 
+func (h *dynUserHandler) getLastUsed(users []*apikey.User) map[string]time.Time {
+	usersWithTime := make(map[string]time.Time, len(users))
+	for _, user := range users {
+		usersWithTime[user.Id] = user.LastUsedAt
+	}
+
+	nodes := h.nodesGetter.Nodes()
+	if len(nodes) == 1 {
+		return usersWithTime
+	}
+
+	// we tolerate errors in requests to other nodes and don't want to wait too long. Last used time is a best-effort
+	// operation
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second)
+	defer cancelFunc()
+	userStatuses := make([]*apikey.UserStatusResponse, len(nodes))
+	wg := &sync.WaitGroup{}
+	wg.Add(len(nodes))
+	for i, nodeName := range nodes {
+		i, nodeName := i, nodeName
+		enterrors.GoWrapper(func() {
+			status, err := h.remoteUser.GetUserStatus(ctx, nodeName, usersWithTime, true)
+			if err == nil {
+				userStatuses[i] = status
+			}
+			wg.Done()
+		}, h.logger)
+	}
+	wg.Wait()
+
+	for _, status := range userStatuses {
+		if status == nil {
+			continue
+		}
+		for userId, lastUsedTime := range status.Users {
+			if lastUsedTime.After(usersWithTime[userId]) {
+				usersWithTime[userId] = lastUsedTime
+			}
+		}
+	}
+
+	// update all other nodes with maximum time so usage does not "jump back" when the node that has the latest time
+	// recorded is down.
+	// This is opportunistic (we dont care about errors) and there is no need to keep the request waiting for this
+	enterrors.GoWrapper(func() {
+		ctx2, cancelFunc2 := context.WithTimeout(context.Background(), time.Second)
+		defer cancelFunc2()
+		wg := &sync.WaitGroup{}
+		wg.Add(len(nodes))
+		for _, nodeName := range nodes {
+			nodeName := nodeName
+			enterrors.GoWrapper(func() {
+				// dont care about returns or errors
+				_, _ = h.remoteUser.GetUserStatus(ctx2, nodeName, usersWithTime, false)
```

Comment:
question: so GetUserStatus also updates last used time?

Reply:
I see; it's a rather unexpected side-effect, but I guess part of the design.
do you think we could call `UpdateLastUsedTimeStamp` directly here?

Reply:
ah, it's defined on a different struct, nvmnd.
still, some comment at call site or / docstring for GetUserStatus would be good.

Reply:
I renamed the function to GetAndUpdateLastUsedTime - it is not just on another struct, but calls other _nodes_